pshtt/utils.py

```diff
@@ -75,3 +75,28 @@ def configure_logging(debug=False):
         log_level = logging.WARNING
 
     logging.basicConfig(format='%(message)s', level=log_level)
+
+
+# Ensure domains given are not prepended with www/https/http/combination of both.
+def format_domain(domain_name):
+    # Check for www. prepended
+    if domain_name.startswith("www."):
+        return domain_name[4:]
+    # Check for https prepended, account for www. as well
+    if domain_name.startswith("https://"):
+        return format_domain(domain_name[8:])
+
+    # Check for http:// prepended, account for www. as well
+    if domain_name.startswith("http://"):
+        return format_domain(domain_name[7:])
```

Comment:
Instead of doing this recursively, I think it would be more predictable and safe to do this iteratively. I also think it would be more readable and clear to couple the detection of the prefix with the modification to the string itself, rather than truncate it using character indexes.

So first it should replace a regex of `^https?://` with an empty string, and then it should replace any remaining prefix of `^www\.` with an empty string. If we also merged this into `utils.format_domains()`, you would then append the result to the ongoing array and then return the whole thing once they're all done.

With the current recursive solution, `http://http://domain.com` would be turned into `domain.com` -- and despite `http://http://domain.com` being a broken domain, that is actually not what what we are trying to fix with this check. We're trying to offer a straightforward convenience to remove a `http://` or `https://` protocol prefix, and a single `www.` subdomain prefix if it exists.